backend/cfg/cfg.ml

```diff
@@ -284,19 +284,24 @@ let dump_op ppf = function
   | Floatop (Float32, op) ->
     Format.fprintf ppf "float32op %a" Printmach.floatop op
   | Csel _ -> Format.fprintf ppf "csel"
-  | Valueofint -> Format.fprintf ppf "valueofint"
-  | Intofvalue -> Format.fprintf ppf "intofvalue"
-  | Vectorcast Bits128 -> Format.fprintf ppf "vec128->vec128"
-  | Scalarcast Float32_as_float -> Format.fprintf ppf "float32 as float"
-  | Scalarcast (Float_of_int Float64) -> Format.fprintf ppf "int->float"
-  | Scalarcast (Float_to_int Float64) -> Format.fprintf ppf "float->int"
-  | Scalarcast (Float_of_int Float32) -> Format.fprintf ppf "int->float32"
-  | Scalarcast (Float_to_int Float32) -> Format.fprintf ppf "float32->int"
-  | Scalarcast Float_of_float32 -> Format.fprintf ppf "float32->float"
-  | Scalarcast Float_to_float32 -> Format.fprintf ppf "float->float32"
-  | Scalarcast (V128_to_scalar ty) ->
+  | Reinterpret_cast V128_of_v128 -> Format.fprintf ppf "vec128 as vec128"
+  | Reinterpret_cast Value_of_int -> Format.fprintf ppf "int as value"
+  | Reinterpret_cast Int_of_value -> Format.fprintf ppf "value as int"
+  | Reinterpret_cast Float32_of_float -> Format.fprintf ppf "float as float32"
+  | Reinterpret_cast Float_of_float32 -> Format.fprintf ppf "float32 as float"
+  | Reinterpret_cast Float_of_int64 -> Format.fprintf ppf "int64 as float"
+  | Reinterpret_cast Int64_of_float -> Format.fprintf ppf "float as int64"
+  | Reinterpret_cast Float32_of_int32 -> Format.fprintf ppf "int32 as float32"
+  | Reinterpret_cast Int32_of_float32 -> Format.fprintf ppf "float32 as int32"
+  | Static_cast (Float_of_int Float64) -> Format.fprintf ppf "int->float"
+  | Static_cast (Int_of_float Float64) -> Format.fprintf ppf "float->int"
+  | Static_cast (Float_of_int Float32) -> Format.fprintf ppf "int->float32"
+  | Static_cast (Int_of_float Float32) -> Format.fprintf ppf "float32->int"
+  | Static_cast Float_of_float32 -> Format.fprintf ppf "float32->float"
```

Comment:
(It is not really part of this pull request,
but I wonder whether operations that
involve rounding should be considered
not pure.)

Reply:
Hmm, that might be true for the `Round_current_f64_i64` SIMD instruction, which uses the current rounding mode.
I think it doesn't apply to these, though, because they always truncate to int / pick nearest float regardless of rounding mode.

Reply:
Indeed, you cannot manipulate it
from OCaml code - I am not sure
there are rules about the C bindings,
but I agree that if someone changes
the rounding mode that way, they
should use `opaque` to forbid any
code motion.

Reply:
Added a cr-someday